src/main/java/ui/components/pickers/LabelPickerDialog.java

```diff
@@ -3,157 +3,260 @@
     }
 
-    private void setupEvents(Stage stage) {
-        setupKeyEvents();
+    // Population of UI elements
 
-        showingProperty().addListener(e -> {
-            positionDialog(stage);
-        });
+    /**
+     * Populate respective panes with labels that matches current user input
+     * @param state
+     */
+    private void populatePanes(LabelPickerState state) {
+        // Population of UI elements
+        populateAssignedLabels(state.getInitialLabels(), state.getRemovedLabels(), state.getAddedLabels(),
+                               state.getCurrentSuggestion());
+        populateFeedbackLabels(state.getAssignedLabels(), state.getMatchedLabels(), state.getCurrentSuggestion());
+        // Ensures dialog pane resize according to content
+        getDialogPane().getScene().getWindow().sizeToScene();
     }
 
-    private void setupKeyEvents() {
-        textField.textProperty().addListener((observable, oldValue, newValue) -> {
-            uiLogic.processTextFieldChange(newValue.toLowerCase());
-        });
-        textField.setOnKeyPressed(e -> {
-            if (!e.isAltDown() && !e.isMetaDown() && !e.isControlDown()) {
-                if (e.getCode() == KeyCode.DOWN) {
-                    e.consume();
-                    uiLogic.moveHighlightOnLabel(true);
-                } else if (e.getCode() == KeyCode.UP) {
-                    e.consume();
-                    uiLogic.moveHighlightOnLabel(false);
-                } else if (e.getCode() == KeyCode.SPACE) {
-                    e.consume();
-                    uiLogic.toggleSelectedLabel(textField.getText());
-                }
+    private void populateAssignedLabels(List<String> initialLabels, List<String> removedLabels,
+                                        List<String> addedLabels, Optional<String> suggestion) {
+        assignedLabels.getChildren().clear();
+        populateInitialLabels(initialLabels, removedLabels, suggestion);
+        populateToBeAddedLabels(addedLabels, suggestion);
+        if (initialLabels.isEmpty()) createTextLabel("No currently selected labels. ");
+    }
+
+    private void populateInitialLabels(List<String> initialLabels, List<String> removedLabels,
+                                       Optional<String> suggestion) {
+        initialLabels.stream()
+            .forEach(label -> assignedLabels.getChildren()
+            .add(processInitialLabel(label, removedLabels, suggestion)));
+    }
+
+    private Node processInitialLabel(String initialLabel, List<String> removedLabels, 
+                                            Optional<String> suggestion) {
+        TurboLabel repoInitialLabel = TurboLabel.getFirstMatchingTurboLabel(repoLabels, initialLabel);
+        if (!removedLabels.contains(initialLabel)) {
+            if (suggestion.isPresent() && initialLabel.equals(suggestion.get())) {
+                return getStyledPickerLabel(repoInitialLabel, true, false, true, false, true);
```

Comment:
Instead of this, maybe we could do `getStyledPickerLabel(repoInitialLabel).ticked().highlighted().faded()`? Since we go through the trouble of defining all those getters and setters, might as well make call sites more readable.